daemon/cluster/listen_addr.go

```diff
@@ -0,0 +1,163 @@
+package cluster
+
+import (
+	"errors"
+	"fmt"
+	"net"
+)
+
+func resolveListenAddr(specifiedAddr string, defaultListenAddr string) (string, error) {
+	specifiedHost, specifiedPort, err := net.SplitHostPort(specifiedAddr)
+	if err != nil {
+		return "", fmt.Errorf("could not parse listen address %s", specifiedAddr)
```

Comment:
I'm fine with not accepting an empty string in API. In API level this can easily mean that user just forgot that this field existed and will likely have incorrect behavior then. Otoh port/host completion is nice because if you are used to specifying port or host only in cli you would not know that these are not supported in API level.